span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java

```diff
@@ -4,36 +4,53 @@
 
 import com.typesafe.config.Config;
 import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import io.micrometer.core.instrument.Counter;
 import org.apache.kafka.streams.KeyValue;
 import org.apache.kafka.streams.kstream.Transformer;
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.hypertrace.core.datamodel.RawSpan;
-import org.hypertrace.core.datamodel.shared.HexUtils;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
 import org.hypertrace.core.spannormalizer.TraceIdentity;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
 public class JaegerSpanToAvroRawSpanTransformer implements
     Transformer<byte[], Span, KeyValue<TraceIdentity, RawSpan>> {
 
   private static final Logger LOGGER = LoggerFactory
       .getLogger(JaegerSpanToAvroRawSpanTransformer.class);
 
+  private static final String SPAN_RECEIVED_COUNT = "span.received.count";
+  private static final String VALID_SPAN_RECEIVED_COUNT = "valid.span.received.count";
+  private final ConcurrentMap<String, Counter> tenantToSpanReceivedCount = new ConcurrentHashMap<>();
+
+  private static Counter spanCounter = null;
+
   private JaegerSpanNormalizer converter;
 
   @Override
   public void init(ProcessorContext context) {
     Config jobConfig = (Config) context.appConfigs().get(SPAN_NORMALIZER_JOB_CONFIG);
     converter = JaegerSpanNormalizer.get(jobConfig);
+    spanCounter = PlatformMetricsRegistry.registerCounter(SPAN_RECEIVED_COUNT, new HashMap<>());
   }
 
   @Override
   public KeyValue<TraceIdentity, RawSpan> transform(byte[] key, Span value) {
     try {
+      //this is total spans count received. Irrespective of the fact we are able to parse them, or they have tenantId or not.
+      spanCounter.increment();
       RawSpan rawSpan = converter.convert(value);
       if (null != rawSpan) {
-        String traceId = HexUtils.getHex(rawSpan.getTraceId());
```

Comment:
@surajpuvvada  this var was unused and we are directly using `rawSpan.getTraceId()` as key to build TraceIdentity. Not sure if this is intentional ?

Reply:
likely missed it. thanks for fixing it